packages/angular/src/generators/scam/lib/create-module.ts

```diff
@@ -83,19 +83,55 @@ export function createScam(tree: Tree, schema: Schema) {
     )}`;
 
     tree.write(componentFilePath, updatedComponentSource);
+    exportScam(tree, schema, componentFilePath);
     return;
   }
 
+  const scamFilePath = joinPathFragments(
+    componentDirectory,
+    `${componentNames.fileName}.module.ts`
+  );
+
   tree.write(
-    joinPathFragments(
-      componentDirectory,
-      `${componentNames.fileName}.module.ts`
-    ),
+    scamFilePath,
     createSeparateAngularComponentModuleFile(
       `${componentNames.className}${typeNames.className}`,
       componentFileName
     )
   );
+
+  exportScam(tree, schema, scamFilePath);
+}
+
+function exportScam(tree: Tree, schema: Schema, scamFilePath: string) {
+  if (!schema.export) {
+    return;
+  }
+
+  const project =
+    schema.project ?? readWorkspaceConfiguration(tree).defaultProject;
+  const { sourceRoot } = readProjectConfiguration(tree, project);
+
+  const projectEntryPoint = joinPathFragments(sourceRoot, `index.ts`);
```

Comment:
You can try to obtain the right path from the `ng-package.json` file, that file has a `lib.entryFile` which is what you need. Buildable and publishable libs have that file with that property specified. For libs without that file (non-buildable), you can fall back to a best guess like you're doing here.

Reply:
Good idea!